### PR TITLE
api.add_variable() add parameter keyword

### LINKED ISSUE #303: api.add_variable() add parameter keyword
Currently it is not possible to associate a newly created variable with a keyword. But this connection between variables and defined keywords is essential to find variables later, so we need to add this functionality to metacatalog.

`api.add_variable(session, name, symbol, keyword)`, where keyword can be the `keyword_id` or the `full_path`.

On the V4W(server) database there are currently also some variables added by upload scripts without keyword linking, where possible I would add keyword ids manually.

### PR DESCRIPTION
**Pull request by AlexDo1**
Adds the optional parameter `keyword` to `api.add_variable()` to make it possible to link a variable to a keyword when a new varible is added.

Closes #303

### CODE REVIEW

File: metacatalog/api/add.py [hunk 1]
```diff
@@ -159,9 +165,20 @@ def add_variable(session: 'Session', name: str, symbol: str, column_names: List[
...
     attrs['unit_id'] = unit.id
 
+    # get the keyword
+    if keyword:
+        if isinstance(keyword, int):
+            keyword = api.find_keyword(session=session, id=keyword, return_iterator=True).one()
+        elif isinstance(keyword, str):
+            keyword = api.find_keyword(session=session, full_path=keyword, return_iterator=True).first()
+        else:
+            raise AttributeError('keyword has to be of type integer or string.')
```
Comment (mmaelicke) [inline metacatalog/api/add.py:178]: Would it be too much work to add a unittest here?

File: metacatalog/api/find.py [hunk 2]
```diff
@@ -133,6 +137,8 @@ def find_keyword(session: 'Session', id: Optional[int] = None, uuid: Optional[st
     # add needed filter
     if id is not None:
         query = query.filter(models.Keyword.id==id)
+    if full_path is not None:
+        query = query.filter(models.Keyword.full_path==full_path)
```
Comment (mmaelicke) [inline metacatalog/api/find.py:141]: This way, the path has to be a perfect match. Is that a good strategy? For other text fields, we implemented a logic that switches to a `LIKE` statement if the filter statement included a ' *' or '%'. 
Doesn't that make sense here as well?

**Review by mmaelicke — APPROVED** [review]
I will approve, but I have a few minor questions

File: metacatalog/api/add.py (hunk 1 above)
Comment (AlexDo1) [reply metacatalog/api/add.py:178]: I added a smal unittest which adds a unit and a variable and finds it afterwards.

**Pull request merged**